docs/content/Caching/Running-in-Production.md

```diff
@@ -0,0 +1,212 @@
+| complete reference, please consult the [Cube Store section of the Environment
+| Variables reference][ref-config-env].
+<!-- prettier-ignore-end -->
+
+Next, run Cube.js and tell it to connect to Cube Store running on `localhost`
+(on the default port `3030`):
+
+```bash
+docker run -p 4000:4000 \
+  -e CUBEJS_CUBESTORE_HOST=localhost \
+  -v ${PWD}:/cube/conf \
+  cubejs/cube
+```
+
+In the command above, we're specifying `CUBEJS_CUBESTORE_HOST` to let Cube.js
+know where Cube Store is running.
+
+You can also use Docker Compose to achieve the same:
+
+```yaml
+version: '2.2'
+services:
+  cubestore:
+    image: cubejs/cubestore:latest
+    environment:
+      - CUBESTORE_REMOTE_DIR=/cube/data
+    volumes:
+      - .cubestore:/cube/data
+
+  cube:
+    image: cubejs/cube:latest
+    ports:
+      - 4000:4000
+    environment:
+      - CUBEJS_CUBESTORE_HOST=localhost
+      - CUBEJS_CUBESTORE_PORT=3030
+    depends_on:
+      - cubestore
+    links:
+      - cubestore
+    volumes:
+      - ./schema:/cube/conf/schema
+```
+
+### Scaling
+
+<!-- prettier-ignore-start -->
+[[warning | ]]
+| Cube Store _can_ be run in a single instance mode, but this is usually
+| unsuitable for production deployments. For high concurrency and data
+| throughput, we **strongly** recommend running Cube Store as a cluster of
+| multiple instances instead.
+<!-- prettier-ignore-end -->
+
+Scaling Cube Store for a higher concurrency is relatively simple when running in
+cluster mode. Because [the storage layer](#running-in-production-storage) is
+decoupled from the query processing engine, you can horizontally scale your Cube
+Store cluster for as much concurrency as you require.
+
+In cluster mode, Cube Store runs two kinds of nodes:
+
+- a single **router** node handles incoming client connections, manages database
+  metadata and serves simple queries.
+- multiple **worker** nodes which execute SQL queries
+
+In terms of configuration, both the router and worker nodes **must** specify the
+`CUBESTORE_SERVER_NAME` environment variable. Additionally, the router node
+**must** specify the `CUBESTORE_META_PORT` and `CUBESTORE_WORKERS` environment
+variables; worker nodes **must** specify the `CUBESTORE_WORKER_PORT` and
+`CUBESTORE_META_ADDR` environment variables. More information about these
+variables can be found [in the Environment Variables reference][ref-config-env].
+
+<!-- prettier-ignore-start -->
+[[info | ]]
+| To fully take advantage of the worker nodes in the cluster, we **strongly**
+| recommend using [partitioned pre-aggregations][ref-caching-partitioning].
+<!-- prettier-ignore-end -->
+
+A sample Docker Compose stack setting this up might look like:
+
+```yaml
+version: '2.2'
+services:
+  cubestore_router:
+    restart: always
+    image: cubejs/cubestore:latest
+    environment:
+      - CUBESTORE_LOG_LEVEL=trace
+      - CUBESTORE_SERVER_NAME=cubestore_router:9999
+      - CUBESTORE_META_PORT=9999
```

Comment:
@hassankhan I feel that `CUBESTORE_LOG_LEVEL`, `CUBESTORE_SERVER_NAME`, `CUBESTORE_META_PORT` are all optional config.